datasource/ethereum/src/block_stream.rs

```diff
@@ -401,145 +306,93 @@ where
-                        // just prior to the reorg threshold. It isn't safe to go any farther
-                        // due to race conditions.
-                        let to_limit = cmp::min(
-                            head_ptr.number - reorg_threshold,
-                            next_start_block - 1,
-                        );
-
-                        let to = if block_filter.trigger_every_block {
-                            // If there is a block trigger on every block, go
-                            // one block at a time.
-                            from
-                        } else {
-                            // Otherwise use ETHEREUM_BLOCK_RANGE_SIZE to ensure the subgraph
-                            //  block ptr is updated frequently.
-                            let speedup = if from < *ETHEREUM_FAST_SCAN_END {
-                                FAST_SCAN_SPEEDUP
+            Box::new(
+                ctx.eth_adapter
+                    .is_on_main_chain(
+                        &ctx.logger,
+                        ctx.metrics.ethrpc_metrics.clone(),
+                        subgraph_ptr,
+                    )
+                    .and_then(
+                        move |is_on_main_chain| -> Box<dyn Future<Item = _, Error = _> + Send> {
+                            if !is_on_main_chain {
+                                // The subgraph ptr points to a block that was uncled.
+                                // We need to revert this block.
+                                return Box::new(future::ok(ReconciliationStep::RevertBlock(
+                                    subgraph_ptr,
+                                )));
+                            }
+
+                            // The subgraph ptr points to a block on the main chain.
+                            // This means that the last block we processed does not need to be
+                            // reverted.
+                            // Therefore, our direction of travel will be forward, towards the
+                            // chain head.
+
+                            // As an optimization, instead of advancing one block, we will use an
+                            // Ethereum RPC call to find the first few blocks that have event(s) we
+                            // are interested in that lie within the block range between the subgraph ptr
+                            // and either the next data source start_block or the reorg threshold.
+                            // Note that we use block numbers here.
+                            // This is an artifact of Ethereum RPC limitations.
+                            // It is only safe to use block numbers because we are beyond the reorg
+                            // threshold.
+
+                            // Start with first block after subgraph ptr
+                            let from = subgraph_ptr.number + 1;
```

Comment:
I know this code was here before, but the name `from` is pretty misleading (especially when looking at the `let to = if block_filter.trigger_every_block { from }` assignment below. What do you think about updating it to `to` or `next_block_num`?

Reply:
I removed that special case, at some point I added it because block triggers were new and I was afraid of them, but they seem to work fine with the normal range.

Reply:
Hehe, ok 😁